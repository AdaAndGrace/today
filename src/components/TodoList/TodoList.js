import React from 'react';
import {StyleSheet} from 'react-native';
import {OptimizedFlatList} from 'react-native-optimized-flatlist'
import ListItem from '../ListItem/ListItem';

export default class TodoList extends React.Component {
    state = {
        refresh: true
    };

    handleUpdateStatus = (id) => {
        this.props.onUpdateStatus(id);
        this.setState(
            {
                refresh: !this.state.refresh
            }
        );
    };

    _renderItem = ({item, index}) => (
        <ListItem
            key={index}
            title={item.title}
            tea={item.tea}
            status={item.status}
            category={item.category}
            id={item.id}
            onHandleUpdateStatus={this.handleUpdateStatus}
        />
    );

    render() {
        return (
            <OptimizedFlatList
                data={this.props.items}
                extraData={this.state.refresh}
                keyExtractor={(item, index) => index}
                renderItem={this._renderItem}
                initialNumToRender={3}
            />
        );
    }
}
