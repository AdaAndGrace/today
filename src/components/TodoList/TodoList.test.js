import React from 'react';
import TodoList from './TodoList';

import renderer from 'react-test-renderer';

const list = [
    {
        title: 'Take out the trash',
        tea: 'task',
        done: false,
        status: 'completed'
    },
    {
        title: 'March for Our Lives',
        tea: 'event',
        done: false,
        status: 'todo',
        category: 'important'
    },
    {
        title: 'Call Amy',
        tea: 'task',
        done: false,
        status: 'todo',
        category: 'inspiration'
    },
    {
        title: 'A really big idea that wraps lines',
        category: 'idea'
    }
    ]


it('renders without crashing', () => {
    const wrapper = renderer.create(<TodoList items={list}/>).toJSON();
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
});

