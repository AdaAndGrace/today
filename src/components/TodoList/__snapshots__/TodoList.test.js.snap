// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
<RCTScrollView
  data={
    Array [
      Object {
        "done": false,
        "status": "completed",
        "tea": "task",
        "title": "Take out the trash",
      },
      Object {
        "category": "important",
        "done": false,
        "status": "todo",
        "tea": "event",
        "title": "March for Our Lives",
      },
      Object {
        "category": "inspiration",
        "done": false,
        "status": "todo",
        "tea": "task",
        "title": "Call Amy",
      },
      Object {
        "category": "idea",
        "title": "A really big idea that wraps lines",
      },
    ]
  }
  disableVirtualization={false}
  extraData={true}
  getItem={[Function]}
  getItemCount={[Function]}
  horizontal={false}
  initialNumToRender={3}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  onViewableItemsChanged={[Function]}
  renderItem={[Function]}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={
    Array [
      Object {
        "onViewableItemsChanged": [Function],
        "viewabilityConfig": undefined,
      },
    ]
  }
  windowSize={21}
>
  <View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginBottom": 3,
              "marginTop": 1,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#aaa",
                    },
                    Object {
                      "fontSize": 26,
                      "marginRight": 5,
                      "width": 36,
                    },
                  ],
                  Object {
                    "fontFamily": "bullets",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Array [
                    Object {
                      "color": "#aaa",
                    },
                    Object {
                      "fontSize": 26,
                      "marginRight": 5,
                      "width": 36,
                    },
                  ],
                  Object {
                    "fontFamily": "bullets",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#aaa",
                    "textDecorationLine": "line-through",
                  },
                  Object {
                    "flexWrap": "wrap",
                    "fontSize": 26,
                    "lineHeight": 30,
                    "paddingRight": 6,
                    "width": 278,
                  },
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontFamily": "WaitingForTheSunrise",
                  }
                }
              >
                Take out the trash
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginBottom": 3,
              "marginTop": 1,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 26,
                    "marginRight": 5,
                    "width": 36,
                  },
                  Object {
                    "fontFamily": "bullets",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 26,
                    "marginRight": 5,
                    "width": 36,
                  },
                  Object {
                    "fontFamily": "bullets",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 26,
                  "lineHeight": 30,
                  "paddingRight": 6,
                  "width": 278,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontFamily": "WaitingForTheSunrise",
                  }
                }
              >
                March for Our Lives
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={null}
    >
      <View
        onLayout={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "flexDirection": "row",
              "marginBottom": 3,
              "marginTop": 1,
            }
          }
        >
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 26,
                    "marginRight": 5,
                    "width": 36,
                  },
                  Object {
                    "fontFamily": "bullets",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View
            accessibilityComponentType={undefined}
            accessibilityLabel={undefined}
            accessibilityTraits={undefined}
            accessible={true}
            collapsable={undefined}
            hasTVPreferredFocus={undefined}
            hitSlop={undefined}
            isTVSelectable={true}
            nativeID={undefined}
            onLayout={undefined}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
            testID={undefined}
            tvParallaxProperties={undefined}
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontSize": 12,
                  },
                  Object {
                    "fontSize": 26,
                    "marginRight": 5,
                    "width": 36,
                  },
                  Object {
                    "fontFamily": "bullets",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                ]
              }
            >
              
            </Text>
          </View>
          <View>
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              style={
                Object {
                  "flexWrap": "wrap",
                  "fontSize": 26,
                  "lineHeight": 30,
                  "paddingRight": 6,
                  "width": 278,
                }
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Object {
                    "fontFamily": "WaitingForTheSunrise",
                  }
                }
              >
                Call Amy
              </Text>
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 0,
        }
      }
    />
  </View>
</RCTScrollView>
`;
